src/redact.js

```diff
@@ -40,36 +40,40 @@
  */
 module.exports = function redact() {
   if (this.Compiler) {
-    const Compiler = this.Compiler;
-    const visitors = Compiler.prototype.visitors;
+    const visitors = this.Compiler.prototype.visitors;
+    const stringifyContent = function(node) {
+      return (node.redactionContent || [])
+        .map(content => this.visit(content, node))
+        .join("");
+    };
 
     let index = 0;
-    let openBlockIndexes = [];
 
-    visitors.redaction = function redaction(node) {
-      let value = "";
-
-      var self = this;
-      var exit = self.enterLink();
-      if (node.children) {
-        value = self.all(node).join('');
+    visitors.inlineRedaction = function(node) {
+      let exit;
+      if (node.redactionType === "link" || node.redactionType === "image") {
+        exit = this.enterLink();
```

Comment:
Why are link and image special cased?

Reply:
So they don't escape the urls; see https://github.com/remarkjs/remark/blob/049b4033aaace271e4559604262b302d84b6234c/packages/remark-stringify/lib/compiler.js#L20-L23